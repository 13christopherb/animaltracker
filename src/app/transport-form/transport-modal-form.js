import React from 'react';
import {Field} from 'redux-form'
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {Button, Col, Form, Modal} from 'react-bootstrap';
import {Input, Select} from "../form-fields/index";
import moment from "moment";

/**
 * Form for creating a new animal in the form of a modal
 * @param handleSubmit
 * @param handleChange
 * @param onSubmit
 * @param props
 * @returns {*}
 * @constructor
 */
export const TransportModalForm = ({handleSubmit, handleChange, onSubmit, toggleModal, show, ...props}) => {
    return (
        <Modal show={show} onHide={toggleModal}>
            <Modal.Header closeButton>
                <Modal.Title>New Transport</Modal.Title>
            </Modal.Header>
            <Form onSubmit={handleSubmit(onSubmit)}>
                <Modal.Body className="modal-body">
                    <Form.Row>
                        <Col xs={6}>
                            <Field
                                component={Select}
                                title="Departs"
                                id="departs"
                                name="departs"
                                value={props.value}
                                onChange={handleChange}
                                options={[
                                    'NRO',
                                    'MBO',
                                    'SLO'
                                ]}
                            />
                        </Col>
                        <Col xs={6}>
                            <Field
                                component={Select}
                                title="Arrives"
                                name="arrives"
                                id="arrives"
                                value={props.value}
                                onChange={handleChange}
                                options={[
                                    'NRO',
                                    'MBO',
                                    'SLO'
                                ]}
                            />
                        </Col>
                    </Form.Row>
                    <Form.Row>
                        <Col xs={12}>
                            <Field
                                component={Input}
                                name="meetPlace"
                                id="meetPlace"
                                title="Meet Place"
                                type="text"
                                value={props.value}
                                onChange={handleChange}
                            />
                        </Col>
                    </Form.Row>
                    <Form.Row>
                        <Col xs={8}>
                            <Field
                                component={Select}
                                name="meetDate"
                                id="meetDate"
                                title="Meet Date"
                                value={props.value}
                                onChange={handleChange}
                                options={[
                                    moment().format('dddd MM/DD'),
                                    moment().add(1, 'days').format('dddd MM/DD'),
                                    moment().add(2, 'days').format('dddd MM/DD'),
                                    moment().add(3, 'days').format('dddd MM/DD'),
                                ]}
                            />
                        </Col>
                    </Form.Row>
                    <Form.Row>
                        <Col xs={6}>
                            <Field
                                component={Input}
                                name="meetTime"
                                id="meetTime"
                                title="Meet Time"
                                type="text"
                                value={props.value}
                                onChange={handleChange}
                            />
                        </Col>
                    </Form.Row>
                </Modal.Body>
                <Modal.Footer>
                    <Button variant="primary" onClick={toggleModal}
                            disabled={props.pristine || !props.valid}
                            type="submit">
                        Submit
                    </Button>
                    <Button variant="danger" onClick={toggleModal}
                            type="button">
                        Cancel
                    </Button>
                </Modal.Footer>
            </Form>
        </Modal>
    );
};