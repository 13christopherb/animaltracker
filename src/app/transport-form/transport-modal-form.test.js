import React from 'react';
import {shallow, mount} from 'enzyme';
import {Button, Col, Form, Modal} from 'react-bootstrap';
import {TransportModalForm} from "./transport-modal-form";
import toJson from 'enzyme-to-json';
import moment from 'moment';

const transport = {
    departs: 'MBO',
    arrives: 'NRO',
    meetTime: moment('2017-09-15 09:30:00') //Arbitrary time
};

describe('<TransportModalForm />', () => {
    const mockSubmit = jest.fn();
    const mockChange = jest.fn();
    const hideModal = jest.fn();
    const defaultTime = moment('2017-09-15 09:30:00') //Arbitrary time

    beforeEach(() => {
        hideModal.mockReset();
        mockSubmit.mockReset();
        mockChange.mockReset();
    });
    describe('render()', () => {
        it('should render the component', () => {
            const wrapper = shallow(<TransportModalForm handleSubmit={mockSubmit}
                                                        handleChange={mockChange}
                                                        hideModal={hideModal}
                                                        id="MBO"
                                                        defaultTime={defaultTime}/>);
            expect(toJson(wrapper)).toMatchSnapshot()
        })
    });
    describe('submit', () => {
        it('should close on closing button', () => {
            const wrapper = shallow(<TransportModalForm handleSubmit={mockSubmit}
                                                        handleChange={mockChange}
                                                        toggleModal={hideModal}
                                                        id="MBO"
                                                        defaultTime={defaultTime}/>);
            wrapper.find({variant: 'danger'}).simulate('click');
            expect(hideModal.mock.calls.length).toBe(1);
        });
        it('should disable button on form errors', () => {
            const wrapper = shallow(<TransportModalForm handleSubmit={mockSubmit}
                                                        handleChange={mockChange}
                                                        hideModal={hideModal}
                                                        id="MBO"
                                                        defaultTime={defaultTime}
                                                        pristine={false}
                                                        errors={{departs: 'Error'}}/>)
            expect(wrapper.find({type: 'submit'}).props().disabled).toEqual(true);
        });
    })
});