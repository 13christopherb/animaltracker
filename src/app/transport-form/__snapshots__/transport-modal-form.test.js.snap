// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransportModalForm /> render() should render the component 1`] = `
<ForwardRef(Bootstrap(Modal))>
  <ForwardRef(Bootstrap(ModalHeader))
    closeButton={true}
  >
    <ForwardRef(Bootstrap(ModalTitle))>
      New Transport
    </ForwardRef(Bootstrap(ModalTitle))>
  </ForwardRef(Bootstrap(ModalHeader))>
  <ForwardRef(Bootstrap(Form))>
    <ForwardRef(Bootstrap(ModalBody))
      className="modal-body"
    >
      <ForwardRef(Bootstrap(FormRow))>
        <ForwardRef(Bootstrap(Col))
          xs={6}
        >
          <Field
            component={[Function]}
            id="departs"
            name="departs"
            onChange={[MockFunction]}
            options={
              Array [
                "NRO",
                "MBO",
                "SLO",
              ]
            }
            title="Departs"
          />
        </ForwardRef(Bootstrap(Col))>
        <ForwardRef(Bootstrap(Col))
          xs={6}
        >
          <Field
            component={[Function]}
            id="arrives"
            name="arrives"
            onChange={[MockFunction]}
            options={
              Array [
                "NRO",
                "MBO",
                "SLO",
              ]
            }
            title="Arrives"
          />
        </ForwardRef(Bootstrap(Col))>
      </ForwardRef(Bootstrap(FormRow))>
      <ForwardRef(Bootstrap(FormRow))>
        <ForwardRef(Bootstrap(Col))
          xs={12}
        >
          <Field
            component={[Function]}
            id="meetPlace"
            name="meetPlace"
            onChange={[MockFunction]}
            title="Meet Place"
            type="text"
          />
        </ForwardRef(Bootstrap(Col))>
      </ForwardRef(Bootstrap(FormRow))>
      <ForwardRef(Bootstrap(FormRow))>
        <ForwardRef(Bootstrap(Col))
          xs={8}
        >
          <Field
            component={[Function]}
            id="meetDate"
            name="meetDate"
            onChange={[MockFunction]}
            options={
              Array [
                "Wednesday 12/12",
                "Thursday 12/13",
                "Friday 12/14",
                "Saturday 12/15",
              ]
            }
            title="Meet Date"
          />
        </ForwardRef(Bootstrap(Col))>
      </ForwardRef(Bootstrap(FormRow))>
      <ForwardRef(Bootstrap(FormRow))>
        <ForwardRef(Bootstrap(Col))
          xs={6}
        >
          <Field
            component={[Function]}
            id="meetTime"
            name="meetTime"
            onChange={[MockFunction]}
            title="Meet Time"
            type="text"
          />
        </ForwardRef(Bootstrap(Col))>
      </ForwardRef(Bootstrap(FormRow))>
    </ForwardRef(Bootstrap(ModalBody))>
    <ForwardRef(Bootstrap(ModalFooter))>
      <ForwardRef(Bootstrap(Button))
        disabled={true}
        type="submit"
        variant="primary"
      >
        Submit
      </ForwardRef(Bootstrap(Button))>
      <ForwardRef(Bootstrap(Button))
        type="button"
        variant="danger"
      >
        Cancel
      </ForwardRef(Bootstrap(Button))>
    </ForwardRef(Bootstrap(ModalFooter))>
  </ForwardRef(Bootstrap(Form))>
</ForwardRef(Bootstrap(Modal))>
`;
